### PR TITLE
CI: add unit-tests job and expand golden UT coverage

### PR DESCRIPTION
**Pull request by zhangqi-chen**
## Summary
- Add a CPU-only `unit-tests` CI job that runs `pytest tests/golden -v`; installs just `pytest` + CPU-only torch wheel, finishes in ~1–2s locally.
- Expand golden unit tests to cover runner's classic `golden_fn` path, `compile_only`, skip-validation, `_backend_for_platform`, and `RunResult.__str__`; add validation edge cases (missing key, shape mismatch, NaN, default tolerances); prune redundant tests in tensor_spec/validation.
- In `tests/golden/conftest.py`, install stub `pypto` / `pypto.ir` / `pypto.runtime` / `pypto.backend` modules when pypto is not importable, so the unit tests run without building the compiler.

### CODE REVIEW

File: tests/golden/test_runner.py [hunk 1]
```diff
@@ -18,8 +18,8 @@
 
 import pytest
 import torch
-from golden import TensorSpec, run
-from golden.runner import _save_tensors
+from golden import RunConfig, TensorSpec, run
```
[the remark this answers is not in the document]
Comment (zhangqi-chen) [reply tests/golden/test_runner.py:21]: Fixed in eec2f1c — hoisted RunResult to the top-level import.

File: tests/golden/test_runner.py [hunk 2]
```diff
@@ -238,5 +238,203 @@ def test_partial_cache_still_fails(self, three_kinds_specs, tmp_path):
...
+        self, three_kinds_specs, tmp_path,
+    ):
+        compiled_dir = tmp_path / "build"
+        compiled_dir.mkdir()
+
+        def fake_execute(work_dir, tensors, **_kwargs):
+            tensors[1][:] = torch.full_like(tensors[1], 9999.0)
+
+        compile_p, exec_p = _patch_compile_and_execute(compiled_dir)
+        fake = _FakeCompiled(compiled_dir)
```
[the remark this answers is not in the document]
Comment (zhangqi-chen) [reply tests/golden/test_runner.py]: Fixed in eec2f1c — removed the dead _patch_compile_and_execute call; the direct patches below are the only thing needed.

File: tests/golden/test_runner.py [hunk 3]
```diff
@@ -238,5 +238,203 @@ def test_partial_cache_still_fails(self, three_kinds_specs, tmp_path):
...
+        with pytest.raises(ValueError, match="Unknown runtime platform"):
+            _backend_for_platform("notaplatform")
+
+
+class TestRunResultStr:
+    """``RunResult.__str__`` formatting — quick regression pins."""
+
+    def test_pass_with_time(self):
+        from golden.runner import RunResult
+        assert str(RunResult(passed=True, execution_time=1.234)) == "PASS (1.23s)"
```
[the remark this answers is not in the document]
Comment (zhangqi-chen) [reply tests/golden/test_runner.py]: Fixed in eec2f1c — removed the local import.

File: tests/golden/test_runner.py [hunk 4]
```diff
@@ -238,5 +238,203 @@ def test_partial_cache_still_fails(self, three_kinds_specs, tmp_path):
...
+
+    def test_fail_with_error_and_time(self):
+        from golden.runner import RunResult
+        s = str(RunResult(passed=False, error="boom", execution_time=0.5))
+        assert s == "FAIL: boom (0.50s)"
```
[the remark this answers is not in the document]
Comment (zhangqi-chen) [reply tests/golden/test_runner.py]: Fixed in eec2f1c — removed the local import.

File: tests/golden/test_runner.py [hunk 5]
```diff
@@ -238,5 +238,203 @@ def test_partial_cache_still_fails(self, three_kinds_specs, tmp_path):
...
+
+    def test_fail_without_error(self):
+        from golden.runner import RunResult
+        assert str(RunResult(passed=False)) == "FAIL"
```
[the remark this answers is not in the document]
Comment (zhangqi-chen) [reply tests/golden/test_runner.py]: Fixed in eec2f1c — removed the local import.

File: tests/golden/test_runner.py [hunk 6]
```diff
@@ -238,5 +238,203 @@ def test_partial_cache_still_fails(self, three_kinds_specs, tmp_path):
...
+        compiled_dir = tmp_path / "build"
+        compiled_dir.mkdir()
+
+        def fake_execute(work_dir, tensors, **_kwargs):
+            tensors[1][:] = torch.full_like(tensors[1], 9999.0)
+
+        compile_p, exec_p = _patch_compile_and_execute(compiled_dir)
+        fake = _FakeCompiled(compiled_dir)
+        with patch("pypto.ir.compile", return_value=fake), \
+             patch("pypto.runtime.execute_compiled", side_effect=fake_execute):
```
[the remark this answers is not in the document]
Comment (zhangqi-chen) [reply tests/golden/test_runner.py]: Fixed in eec2f1c — dropped the unused _patch_compile_and_execute call.
